plugins/base/src/test/kotlin/translators/DefaultPsiToDocumentableTranslatorTest.kt

```diff
@@ -741,4 +741,93 @@ class DefaultPsiToDocumentableTranslatorTest : BaseAbstractTest() {
+            configuration
+        ) {
+            documentablesMergingStage = { module ->
+                val testedClass = module.findClasslike(packageName = "test", "A") as DClass
+
+                assertEquals(1, testedClass.constructors.size, "Expect 1 default constructor")
+                assertTrue(
+                    testedClass.constructors.first().parameters.isEmpty(),
+                    "Expect default constructor doesn't have params"
+                )
+                assertEquals(JavaVisibility.Public, testedClass.constructors.first().constructorVisibility())
+            }
+        }
+    }
+
+    @Test
+    fun `should have package-private default constructor in package-private class`() {
+        testInline(
+            """
+            |/src/main/java/test/A.java
+            |package test;
+            |class A {
+            |}
+        """.trimIndent(),
+            configuration
+        ) {
+            documentablesMergingStage = { module ->
+                val testedClass = module.findClasslike(packageName = "test", "A") as DClass
+
+                assertEquals(1, testedClass.constructors.size, "Expect 1 default constructor")
+                assertEquals(JavaVisibility.Default, testedClass.constructors.first().constructorVisibility())
+            }
+        }
+    }
+
+    @Test
+    fun `should have private default constructor in private nested class`() {
+        testInline(
+            """
+            |/src/main/java/test/A.java
+            |package test;
+            |class A {
+            |    private static class PrivateNested{}
+            |}
+        """.trimIndent(),
+            configuration
+        ) {
+            documentablesMergingStage = { module ->
+                val parentClass = module.findClasslike(packageName = "test", "A") as DClass
+                val testedClass = parentClass.classlikes.single { it.name == "PrivateNested" } as DClass
+
+                assertEquals(1, testedClass.constructors.size, "Expect 1 default constructor")
+                assertEquals(JavaVisibility.Private, testedClass.constructors.first().constructorVisibility())
+            }
+        }
+    }
+
+    @Test
+    fun `should not have a default constructor because have explicit private`() {
+        testInline(
+            """
+            |/src/main/java/test/A.java
+            |package test;
+            |public class A {
+            |    private A(){}
+            |}
+        """.trimIndent(),
+            configuration
+        ) {
+            documentablesMergingStage = { module ->
+                val testedClass = module.findClasslike(packageName = "test", "A") as DClass
+
+                assertEquals(1, testedClass.constructors.size, "Expect 1 declared constructor")
+                assertEquals(JavaVisibility.Private, testedClass.constructors.first().constructorVisibility())
+            }
+        }
+    }
 }
+
+private fun DFunction.constructorVisibility() = visibility.first().value
```

Comment:
Same thing about scope - it's an extension for any function, there's nothing limiting it to constructors only as far as I can see

So I would change it to just `DFunction.visibility()` or `DClass.constructorVisibility()`